lib/counter_culture.rb

```diff
@@ -286,7 +286,11 @@ def change_counter_cache(options)
       if id_to_change && options[:counter_column]
         delta_magnitude = if options[:delta_column]
                             delta_attr_name = options[:was] ? "#{options[:delta_column]}_was" : options[:delta_column]
-                            self.send(delta_attr_name) || 0
+                            val = self.send(delta_attr_name)
+                            # check if value is a string. This can happen if the
+                            # delta_column is an enum
+                            val = val.is_a?(String) ? self.class.send(options[:delta_column].pluralize)[val] : val
```

Comment:
What does `read_attribute` return for an enum? With a little luck that returns the integer value. :) Then we could just do `self.read_attribute(delta_attr_name)` instead of the `send`.